src/Renderer/Shader/TileFS.glsl

```diff
@@ -122,21 +122,27 @@ void main() {
                             textureIndex,
                             projWGS84 ? vUv_WGS84 : uvPM);
 
-                        if (layerColor.a > 0.0) {
+                        bool isVisible = layerColor.a > 0.0;
+
+                        // if first texture is transparent, we need to force blending with CWhite
+                        if (isVisible || (!isVisible && !validTexture)) {
```

Comment:
Can be simplified to `isVisible || !validTexture`.
But it still doesn't make sense sorry:
- your comment mentions `CWhite` but it's only used if `paramsA.z > 2.0`?
- why do you do anything if layerColor.a is 0?
- if the problem occurs when blending a non-opaque texture with no previous texture, maybe a simple fix would be to do no blending at all? (eg: diffuseColor = layerColor)

If you have a problem when you need to blend the 1st valid texture

Reply:
> your comment mentions CWhite but it's only used if paramsA.z > 2.0?

it's to force the blending with the background first background color (CWhite)
I change comment

> why do you do anything if layerColor.a is 0?

because the `mix` function doesn't make any changes

> if the problem occurs when blending a non-opaque texture with no previous texture, maybe a simple fix would be to do no blending at all? (eg: diffuseColor = layerColor)

but what should be the background color if the opacity (texture or layer) is less than 1.0?

Reply:
> but what should be the background color if the opacity (texture or layer) is less than 1.0?

You have N layers and you want to blend them together, so my guess is the algorithm is supposed to look like this:

The initial color can be either:
  - `noTextureColor`
  - `vec4(0, 0, 0, 0)` (which thanks to premultiplied alpha blending won't affect the final result)

Then for each layer: 
   layer_color = colorAtIdUv(...);
   // premultiplied alpha blending
   diffuseColor = diffuseColor * (1 - layer_color.a * paramsA.w) + layer_color * paramsA.w)

Does this answer your question?

Reply:
in your proposal
`noTextureColor` is equal to `vec4(0, 0, 0, 0)`?

Reply:
Yes, exactly. 
I separated , unassociated the alpha (layerColor.rgb /= layerColor.a), to use `mix` function.
I use your equivalent formulation.

The background color is `noTextureColor`, the user assume the background colour.